ctapipe/io/pointing.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any
+
+import astropy.units as u
+import numpy as np
+import tables
+from scipy.interpolate import interp1d
+
+from ctapipe.core import Component, traits
+
+from .astropy_helpers import read_table
+
+
+class PointingInterpolator(Component):
+    bounds_error = traits.Bool(default_value=True).tag(config=True)
+    extrapolate = traits.Bool(default_value=False).tag(config=True)
+
+    def __init__(self, h5file, **kwargs):
+        super().__init__(**kwargs)
+
+        if not isinstance(h5file, tables.File):
+            raise TypeError("h5file must be a tables.File")
+        self.h5file = h5file
+
+        self.interp_options: dict[str, Any] = dict(assume_sorted=True, copy=False)
+        if self.bounds_error:
+            self.interp_options["bounds_error"] = True
+        elif self.extrapolate:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = "extrapolate"
+        else:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = np.nan
+
+        self._alt_interpolators = {}
+        self._az_interpolators = {}
+
+    def _read_pointing_table(self, tel_id):
+        pointing_table = read_table(
+            self.h5file,
+            f"/dl0/monitoring/telescope/pointing/tel_{tel_id:03d}",
+        )
+
+        # sort first, so it's not done twice for each interpolator
+        pointing_table.sort("time")
+        mjd = pointing_table["time"].tai.mjd
+
+        az = pointing_table["azimuth"].quantity.to_value(u.rad)
+        # prepare azimuth for interpolation "unwrapping", i.e. turning 359, 1 into 359, 361
+        az = np.unwrap(az)
+        alt = pointing_table["altitude"].quantity.to_value(u.rad)
+
+        self._az_interpolators[tel_id] = interp1d(mjd, az, **self.interp_options)
```

Comment:
scipy docs:

>  [interp1d](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html#scipy.interpolate.interp1d) is considered legacy API and is not recommended for use in new code. Consider using more specific interpolators instead.

Reply:
But I don't see a direct "new" replacement for the case of linear interpolation.

I'd stick with it for now.

Reply:
Yes, but that is lacking all the options for how to deal with values outside the interpolation range and it requires passing the data in everytime you call it.